### PR TITLE
Updates on patient module

### PR DESCRIPTION
**Pull request by Arnoldosmium**
Will provide more comment on why some edition happened

Would love to discuss more

### CODE REVIEW

File: api/patient.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 """
-@api {post} /patient/:user_id/create Create patient
+@api {post} /patient/create Create patient
```
Comment (Arnoldosmium) [inline api/patient.py]: Factor out `patient` class by itself. `/patient/:user_id` does not quite make sense...

File: api/patient.py [hunk 2]
```diff
@@ -8,39 +8,55 @@
 @apiParam {String}   last_name			Mandatory last name.
 @apiParam {String}   first_name_pinyin	Mandatory first name pinyin.
 @apiParam {String}   last_name_pinyin	Mandatory last name pinyin.
-@apiParam {String}   gender="M"			Mandatory gender with default "M".
+@apiParam {String}   gender				Mandatory gender with default "M".
```
Comment (Arnoldosmium) [inline api/patient.py]: I would suggest mandatory vars have no default value.

File: api/patient.py [hunk 3]
```diff
@@ -8,39 +8,55 @@
...
 @apiParam {DateTime} birthdate			Mandatory birthdate.
 @apiParam {String}   relationship		Mandatory replationship with user.
 @apiParam {String}	 passport_number 	Mandatory passport number.
-@apiParam {String}	 note				Optional note.
+@apiParam {String}	 [note]				Optional note.
```
Comment (Arnoldosmium) [inline api/patient.py]: trying api_doc api.

File: api/patient.py [hunk 4]
```diff
@@ -8,39 +8,55 @@
...
 
-@apiSuccess {String} msg		Successfully create patient. 
+@apiSuccess {String} patient_id			On Successful creation, return patient_id.
```
Comment (Arnoldosmium) [inline api/patient.py]: Most of the time better return sth rather than just the message

File: api/patient.py [hunk 5]
```diff
@@ -8,39 +8,55 @@
...
 @apiVersion 0.0.0
 """
 
 """
-@api {post} /patient/:user_id/update Update patient
+@api {post} /patient/:patient_id/update Update patient
 @apiName UpdatePatient
 @apiGroup Patient
 
-@apiParam {Number}   user_id			Mandatory unique User ID.
```
Comment (Arnoldosmium) [inline api/patient.py]: We should not expose the update of associating `user_id` in the api

File: api/patient.py [hunk 6]
```diff
@@ -8,39 +8,55 @@
...
-@apiParam {String}   last_name			Mandatory last name.
-@apiParam {String}   first_name_pinyin	Mandatory first name pinyin.
-@apiParam {String}   last_name_pinyin	Mandatory last name pinyin.
-@apiParam {String}   gender="M"			Mandatory gender with default "M".
-@apiParam {DateTime} birthdate			Mandatory birthdate.
-@apiParam {String}   relationship		Mandatory replationship with user.
-@apiParam {String}	 passport_number 	Mandatory passport number.
-@apiParam {String}	 note				Optional note.
+@apiParam {Number}	 patient_id				Mandatory unique Patient ID.
+@apiParam {String}   [first_name]	 		Mandatory first name.
```
Comment (Arnoldosmium) [inline api/patient.py]: These fields could be optional in update endpoint. -- Although the frontend will always include all of them in the update post request.

File: api/patient.py [hunk 7]
```diff
@@ -8,39 +8,55 @@
...
+@apiParam {String}	 [passport_number] 		Mandatory passport number.
+@apiParam {String}	 [note]					Optional note.
 
 @apiSuccess {String} msg		Successfully update patient. 
 
 @apiVersion 0.0.0
 """
 
+"""
+@api {post} /patient/:patient_id/visit_history/add	Create patient previous hospital visit history
```
Comment (Arnoldosmium) [inline api/patient.py]: Adding this endpoint for the page 4.1.2(2).
Several fields like 首诊医院 are not included in the previous endpoints. Maybe should factor out the history as a `visit_history` object

**Comment by jiashenC** [comment]
@gabrielwry review and merge the PR.

**Pull request merged**